tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -3867,6 +3867,28 @@ OpFoldResult SelectOp::fold(ArrayRef<Attribute> operands) {
   return {};
 }
 
+// simplify select(not(%pred), true_value, false_value) => select(%pred,
+// false_value, true_value)
+static LogicalResult selectCanonicalization(SelectOp selectOp,
+                                            PatternRewriter& rewriter) {
+  if (auto notOp = selectOp.pred().getDefiningOp<NotOp>()) {
+    if (1 ==
```

Comment:
It is a requirement of SelectOp that this be i1, so you don't need to check this as this would have been verified.